### DISCUSSION TITLE
metric type not match: expected=IP, actual=L2: invalid parameter

### DISCUSSION THREAD

**Discussion by firstendIsMe** [milvus-io/milvus#27142] [Q&A and General discussion]
i create index :
    collection = Collection(COLLECTION_NAME)
    index_param = {
        'metric_type': 'L2',
        'index_type': 'FLAT'
    }
    collection.create_index(field_name="embedding", index_param=index_param)

but when i search:
    search_params = {"metric_type": "L2"}
    results = collection.search(
        data=[embedding],
        anns_field='embedding',
        param=search_params,
        output_fields=['img_id', 'img_url'],
        limit=10
    )

i got an error result : 
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=All attempts results:
attempt #1:code: UnexpectedError, error: fail to Search, QueryNode ID=10, reason=collection:444223000022469076, metric type not match: expected=IP, actual=L2: invalid parameter

Why is that?  Can anyone help me?

**Comment by firstendIsMe** [comment]
您发送的邮件已收到，谢谢

**Comment by yhmo** [comment]
`collection.create_index(field_name="embedding", index_param=index_param)`

should be:

`collection.create_index(field_name="embedding", index_params=index_param)`
